app/services/data_migrations/resolve_all_fraudulent_duplicate_matches.rb

```diff
@@ -0,0 +1,12 @@
+module DataMigrations
+  class ResolveAllFraudulentDuplicateMatches
+    TIMESTAMP = 20220126171018
+    MANUAL_RUN = false
+
+    def change
+      FraudMatch.all.each do |fraud_match|
+        fraud_match.update(resolved: true) if fraud_match.fraudulent?
+      end
```

Comment:
This should be quicker to run?
```suggestion
      FraudMatch.where(fraud_match: true).update_all(resolved: true)
```

Reply:
Maybe it doesn't matter but I am not sure whether `update_all` will trigger an audit trail entry. If there was a larger number of existing fraud matches we might also want to use `find_each` rather than `each` to stop too many AR objects be instantiated at once.

Reply:
Ah no, [`update_all`](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-update_all) generates a single SQL statement and doesn't trigger any model instantiation or callbacks.

That said, does saving the FraudMatch models generate audit trail entries anyway? It doesn't have an `audited` method on it, but maybe it happens on the candidate models that it joins with?

Reply:
`FraudMatch` is audited, though I'm not sure we display the history anywhere.

Reply:
Interesting 🤔 . All suggestions are good. There are almost 500 records in the fraud matches table so I think we can use `#each` (@avinhurry  but it is good for you to learn about the `#find_each` that Steve mentioned.

**If you wanna trigger the audit**, what I suggest as a final result:

```ruby
  FraudMatch.where(fraudulent: true).each do |fraud_match|
    fraud_match.update(resolved: true)
  end
```

@avinhurry @stevehook @frankieroberto wdyt? ☝️

Reply:
@stevehook ah, so it is! I missed that somehow.

@tomas-stefano your code looks good to me. In theory it might be better to use `find_each` rather than `each`, so that the find queries and record initialisations get batched, but at our scale it's probably irrelevant as we have less than 1000 anyway?

Reply:
Thanks everyone! Have ammended